eZ/Bundle/EzPublishMigrationBundle/Command/LegacyStorage/UpdateSortKeysCommand.php

```diff
@@ -0,0 +1,274 @@
+
+EOT
+            );
+    }
+
+    protected function execute( InputInterface $input, OutputInterface $output )
+    {
+        $fieldTypeIdentifier = $input->getArgument( 'fieldtype_identifier' );
+        $bulkCount = $input->getArgument( 'bulk_count' );
+        $dryRun = $input->getOption( 'dry-run' );
+
+        /** @var \eZ\Publish\Core\Persistence\Database\DatabaseHandler $databaseHandler */
+        $databaseHandler = $this->getContainer()->get( 'ezpublish.connection' );
+
+        /** @var \eZ\Publish\API\Repository\Repository $repository */
+        $repository = $this->getContainer()->get( "ezpublish.api.repository" );
+        $repository->setCurrentUser(
+            $repository->getUserService()->loadUser( 14 )
+        );
+
+        if ( !$this->getContainer()->has( "ezpublish.fieldtype.{$fieldTypeIdentifier}" ) )
+        {
+            $output->writeln( "<error>Field type '{$fieldTypeIdentifier}' was not found.</error> " );
+            return;
+        }
+
+        $warningStyle = new OutputFormatterStyle( "red" );
+        $output->getFormatter()->setStyle( "warning", $warningStyle );
+        $output->writeln( $this->getProcessedHelp() );
+
+        $query = $databaseHandler->createSelectQuery();
+        $query
+            ->select(
+                $query->expr->count(
+                    "DISTINCT " . $query->expr->concat(
+                        $databaseHandler->quoteColumn( "contentobject_id" ),
+                        $query->bindValue( "-", null, PDO::PARAM_STR ),
+                        $databaseHandler->quoteColumn( "version" )
+                    )
+                )
+            )
+            ->from( "ezcontentobject_attribute" )
+            ->where(
+                $query->expr->eq(
+                    $databaseHandler->quoteColumn( "data_type_string" ),
+                    $query->bindValue( $fieldTypeIdentifier )
+                )
+            );
+
+        $stmt = $query->prepare();
+        $stmt->execute();
+        $totalCount = $stmt->fetchColumn();
+
+        $output->writeln(
+            array(
+                "Found total Content versions to update: " . $totalCount,
+                "",
+            )
+        );
+
+        $helper = $this->getHelper( "question" );
+        $question = new ConfirmationQuestion(
+            "<question>Are you sure you want to proceed?</question> ",
+            false
+        );
+
+        if ( !$helper->ask( $input, $output, $question ) )
+        {
+            $output->writeln( "" );
+            return;
+        }
+
+        $query = $databaseHandler->createSelectQuery();
+        $query
+            ->selectDistinct( "contentobject_id, version" )
+            ->from( "ezcontentobject_attribute" )
+            ->where(
+                $query->expr->eq(
+                    $databaseHandler->quoteColumn( "data_type_string" ),
+                    $query->bindValue( $fieldTypeIdentifier )
```

Comment:
Does this fetch all rows from ezcontentobject_attribute of the given data type string, without limits? If so, could be a problem. Tends to be crowded, that table.